server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -132,7 +137,19 @@ SearchRequest transformRequest(SearchRequest request) throws SearchPipelineProce
                         request = processor.processRequest(request);
                     } catch (Exception e) {
                         onRequestProcessorFailed(processor);
-                        throw e;
+                        if (processor.isIgnoreFailure()) {
+                            logger.warn(
```

Comment:
Is warn the appropriate log level here? Given this example use case:

> This pipeline is aiming at standardizing a field migration that named "product_1" before and now renamed to "product_name" and filter out internal customer orders.

It seems like the "failure" case could be the norm, and could happen at an extremely high rate if this is going to throw for every document it sees that doesn't have the field it wishes to rename. Is that right? If so, then logging the stack traces at warn level will create a ton of log output.

Reply:
in a renameProcessor, we have attribute call ignoreMissing [here](https://github.com/opensearch-project/OpenSearch/blob/a39f60f482c405a102e6eec83097e91acc90a889/modules/ingest-common/src/main/java/org/opensearch/ingest/common/RenameProcessor.java#L76). 
If setting ignoreMissing is true and the global ignoreFailure is true, it won't throw out exceptions or log any warn messages. 
But if setting ignoreMissing is false and the global ignoreFailure is true, it will catch the exception and log warn level.

Reply:
Ok, makes sense. This is good as long as "ignoreFailures" is truly meant to catch exceptional cases, which appears to be the case.

Reply:
@mingshl you are talking about a case that is specific to 'renameProcessor', but that won't hold true for all Processors?  The logic in this Pipeline class needs to hold for all Processor implementations, no?